src/components/dropdown/menu.tsx

```diff
@@ -132,7 +132,10 @@ class Menu<T> extends Component<Props<T>> {
             <li
               data-testid={item.name}
               {...getItemProps({
-                key: item.key || item.value?.toString() || `item-${index}`,
+                key:
+                  item.key ||
+                  (typeof item.value !== "object" && item.value.toString()) ||
+                  `item-${index}`,
```

Comment:
the last option seems to always work. do we really need a fallback chain of 3 options?

Reply:
are you suggesting to always use `item-${index}` ?

Reply:
yes. we just need a unique key per item and that seems to do the trick? is there anything I'm missing? what if multiple of those are on the page?

Reply:
yes. basically it's only a react thing. Multiple keys that are the same are throwing a warning. That is what's happening at the moment

Reply:
https://reactjs.org/docs/lists-and-keys.html#keys

you should not use the index as a key if the order of the items change and this might negatively impact the performance

Reply:
As discussed in Slack, since we are using the `<Menu />` component also for the selects with dynamic data, I'll keep the approach as it is